libminifi/include/c2/C2Agent.h

```diff
@@ -171,13 +159,20 @@ class C2Agent : public state::UpdateController, public state::response::Response
    */
   bool update_property(const std::string &property_name, const std::string &property_value,  bool persist = false);
 
+  /**
+   * Creates configuration options C2 payload for response
+   */
+  C2Payload prepareConfigurationOptions(const C2ContentResponse &resp) const;
+
+  void execute(const std::function<state::Update()>& function);
+
   std::timed_mutex metrics_mutex_;
   std::map<std::string, std::shared_ptr<state::response::ResponseNode>> metrics_map_;
 
   /**
-     * Device information stored in the metrics format
-     */
-    std::map<std::string, std::shared_ptr<state::response::ResponseNode>> root_response_nodes_;
+   * Device information stored in the metrics format
+   */
+  std::map<std::string, std::shared_ptr<state::response::ResponseNode>> root_response_nodes_;
```

Comment:
Wonder if we could store unique ptrs in this as well? I know this is old code, so in case it has a bigger impact, I'm ok with a follow-up as well.